### PR TITLE
core: fix INFO() and ERROR() macros in core/serial_ftdi.cpp

### PR DESCRIPTION
**Pull request by bstoeger**
Error introduced in da7ea17b66: the INDO() and ERROR() macros pass stdout instead of the format string as first parameter to report_error(). Ooooops. How did this ever pass the compile tests!?

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Stupid bug - how did this ever compile on Windows!?

### CODE REVIEW

File: core/serial_ftdi.c [hunk 1]
```diff
@@ -38,8 +38,8 @@
 #endif
 
 #include "errorhelper.h"
-#define INFO(context, fmt, ...)	report_info(stderr, "INFO: " fmt, ##__VA_ARGS__)
-#define ERROR(context, fmt, ...)	report_info(stderr, "ERROR: " fmt, ##__VA_ARGS__)
+#define INFO(context, fmt, ...)	report_info("INFO: " fmt, ##__VA_ARGS__)
+#define ERROR(context, fmt, ...)	report_info("ERROR: " fmt, ##__VA_ARGS__)
```
Comment (mikeller) [inline core/serial_ftdi.c:42]: While the bugfix looks good, should we as well get rid of the `context` parameter where it still exists, seeing that this is not used any more? This should probably be done in a separate pull request though.

**Review by mikeller — APPROVED** [review]
Looks good to me, well spotted.

**Comment by bstoeger** [comment]
> well spotted.

Not be me though, the compiler refused once I turned the file to C++.  🤷

File: core/serial_ftdi.c (hunk 1 above)
Comment (bstoeger) [reply core/serial_ftdi.c:42]: I wonder if this parameter ever was used?

Comment (mikeller) [reply core/serial_ftdi.c:42]: I think that's a pattern from libdivecomputer, where the context is providing the sink for the log data.

Comment (glance-) [reply core/serial_ftdi.c:42]: This code grew up in libdivecomputer but ended up in subsurface.

I'm not even sure if this code is in use in any of our builds any more, since we moved to the jni-wrapped java serial drivers on android.
